retroshare-gui/src/gui/gxsforums/GxsForumThreadWidget.cpp

```diff
@@ -2542,26 +2543,129 @@ bool GxsForumThreadWidget::filterItem(QTreeWidgetItem *item, const QString &text
 /** Request / Response of Data ********************************/
 /*********************** **** **** **** ***********************/
 
+void GxsForumThreadWidget::updateGroupData()
+{
+	mSubscribeFlags = 0;
+	mSignFlags = 0;
+	mForumDescription.clear();
+
+	emit groupChanged(this);
+
+	RsThread::async([this]()
+	{
+        // 1 - get message data from p3GxsForums
+
+        std::list<RsGxsGroupId> forumIds;
+		std::vector<RsGxsForumGroup> groups;
+
+        forumIds.push_back(groupId());
+
+		if(!rsGxsForums->getForumsInfo(forumIds,groups))
+		{
+			std::cerr << __PRETTY_FUNCTION__ << " failed to retrieve forum group info for forum " << groupId() << std::endl;
+			return;
+        }
+
+        if(groups.size() != 1)
+        {
+			mStateHelper->setActive(mTokenTypeGroupData, false);
+			mStateHelper->clear(mTokenTypeGroupData);
+			return;
+        }
+
+        // 2 - sort the messages into a proper hierarchy
+
+        RsGxsForumGroup group = groups[0];
+
+        // 3 - update the model in the UI thread.
+
+        RsQThreadUtils::postToObject( [group,this]()
+		{
+			/* Here it goes any code you want to be executed on the Qt Gui
+			 * thread, for example to update the data model with new information
+			 * after a blocking call to RetroShare API complete, note that
+			 * Qt::QueuedConnection is important!

```

Comment:
The QueuedConnection part of the comment is not relevant anymore, so it should be removed ;)